lib/zeitwerk/loader.rb

```diff
@@ -536,9 +536,20 @@ def all_dirs
       autoloads[abspath] = cref
       Registry.register_autoload(self, abspath)
 
-      # See why in the documentation of Zeitwerk::Registry.inceptions.
-      unless cref.autoload?
-        Registry.register_inception(cref.path, abspath, self)
+      # If manual incepted namespaces are present we skip the default behavior
+      # and we will only incept the manually defined namespaces for improved
+      # performance
+      if manual_incepted_namespaces.any?
```

Comment:
May I suggest [`empty?`](https://rubyapi.org/o/set#method-i-empty-3F) here as perhaps a more idiomatic way to check if a set's empty?